src/backend/distributed/commands/multi_copy.c

```diff
@@ -1195,6 +1233,195 @@ AppendCopyBinaryFooters(CopyOutState footerOutputState)
 }
 
 
+/*
+ * CopyToAppendPartitionedTable implements the COPY table_name FROM ... for
+ * append-partitioned tables.
+ */
+static void
+CopyToAppendPartitionedTable(CopyStmt *copyStatement, char *completionTag)
```

Comment:
I think it would be better to put this next to CopyToHashOrRangePartitionedTable